examples/producer_example/producer_example.go

```diff
@@ -44,26 +46,63 @@ func main() {
 
 	fmt.Printf("Created Producer %v\n", p)
 
-	// Optional delivery channel, if not specified the Producer object's
-	// .Events channel is used.
-	deliveryChan := make(chan kafka.Event)
+	// Listen to all the events on the default events channel
+	go func() {
+		for e := range p.Events() {
+			switch ev := e.(type) {
+			case *kafka.Message:
+				m := ev
```

Comment:
Would be good with a comment here saying that this is a message delivery report, indicating success or permanent failure after retries have been exhausted.
Perhaps also mention that application-level retries will typically not help since the client has already done that.